.github/workflows/test.yml

```diff
@@ -65,6 +65,7 @@ jobs:
       matrix:
         python:
           - "3.11"
+          - "3.12"
```

Comment:
This PR makes sense, but I thought we didn’t intend to support Python 3.12 at this time?

Reply:
Au contraire, we have pressing requests to support python 3.12 ASAP. Including from the downstream packaging team.

Reply:
This change does not have to land today, but we need it by the end of the month.

Downstream team requires a release of InstructLab with Python 3.12 within the next few weeks. We need to move to Python 3.12 before we update to RHEL 9.6. One vendor for AI hardware is dropping Python 3.11 support by end of May, too.

Reply:
Sounds good. Thanks for the context! LGTM